Vault docs

alikins opened the pull request:
##### SUMMARY
Start of adding some more vault docs. 

Currently info about 'encrypt_string' and some notes on the vault format.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->

 - Docs Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.4.0 (vault_docs 7e32f31e81) last updated 2017/07/18 11:31:08 (GMT -400)
  config file = None
  configured module search path = [u'/home/adrian/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adrian/src/ansible/lib/ansible
  executable location = /home/adrian/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

alikins committed:
start rst docs for ansible-vault encrypt_string

alikins committed:
wip, vault format docs

(sorta markdown ish atm)

alikins committed:
wip - formatting

alikins committed:
wip, vault docs

alikins committed:
Fix refs to other docs for now

fixing default_role in conf.py will remove need
for this

acrewdson commented:
Since this hasn't been merged yet, I'm wondering: what about adding some example `ansible-vault encrypt_string` command lines to show the usage (which exist in the current Vault docs page)? We could also show an example of how to decrypt an `encrypt_string`-encrypted string using `ansible-vault decrypt`, since that can be non-obvious for someone new to using Vault. I'd be glad to help with either of these. Thank you!

alikins committed:
add 'ref' to cli names

alikins committed:
more vault docs

alikins committed:
wip, misc fixes

alikins committed:
add some encrypt_string examples

alikins committed:
Fix up rstcheck warnings

The code blocks in question included the output that would
be echo'ed from running the command, which isnt valid bash.

alikins committed:
fix formatting and rstcheck warnings about code-block

alikins committed:
Add envvar ref for ANSIBLE_VAULT_PASSWORD_FILE

alikins committed:
fix doc title

dharmabumstead committed:
Fixed title underline

dharmabumstead commented:
Thanks @alikins!

alikins commented:
This Closes #28012
